### PR TITLE
Add Contribution Guide to the project

### PR DESCRIPTION
**Pull request by guettomusick**
# Description
Include a contribution guide on the project

Fixes https://github.com/pokt-network/pocket/issues/34

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Documentation update

# How Has This Been Tested?

No test required

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/contributing/contribution/development.md
Comment (Olshansk) [inline docs/contributing/contribution/development.md:7]: Forking & Coding leads to implementation.

I don't fully understand what `comment` was intending here?

File: docs/contributing/contribution/development.md
Comment (Olshansk) [inline docs/contributing/contribution/development.md:8]: We already have a `code` step between `fork` and `implement`. Should this be `no` given that the other option is `yes`?

**Review by Olshansk — CHANGES_REQUESTED** [review]
I'm about halfway through development.md.

I left a few comments but mainly committed most of my changes directly to my branch. Please see my commits if you're curious to see the changes.

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/integration-issue.md:18]: Sounds good

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/integration-issue.md:13]: That's a good question, @luyzdeleon @andrewnguyen22? I was following a similar goal chat framework here, but does it make sense to keep it as "Origin Document"

File: docs/contributing/contribution/development.md
Comment (guettomusick) [reply docs/contributing/contribution/development.md:7]: The idea was to refer to journaling, but it's not clear. Like you implement something, and you do journaling about it. Maybe I should add a box there.

File: docs/contributing/contribution/development.md
Comment (guettomusick) [reply docs/contributing/contribution/development.md:8]: You are right, it's confusing. let me push a review of this

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/integration-issue.md]: Check 4316cde

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/integration-issue.md:28]: Check 4316cde

File: .github/ISSUE_TEMPLATE/issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/issue.md:1]: Check 4316cde

File: docs/contributing/contribution/development.md
Comment (guettomusick) [reply docs/contributing/contribution/development.md]: Check 4316cde

File: docs/contributing/contribution/development.md
Comment (guettomusick) [reply docs/contributing/contribution/development.md]: Check 4316cde

**Comment by guettomusick** [comment]
@Olshansk I push a new Commit, 4316cde with all but one resolved, please check the fix. About the "origin document" let's wait on Andrew on Luis.

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (luyzdeleon) [reply .github/ISSUE_TEMPLATE/integration-issue.md:13]: Well, the origin document was meant to be both a justification and the planning doc for the effort.

File: docs/contributing/contribution/issue.md
Comment (Olshansk) [inline docs/contributing/contribution/issue.md:52]: IMO, I don't think it's worth embedding the template in the actual doc, but rather link to the markdown doc that is used as the template. That way, if the markdown is updated, this won't be out of date.

"The src code is the source of truth" type of philosophy.

File: docs/contributing/contribution/merge.md
Comment (Olshansk) [inline docs/contributing/contribution/merge.md:34]: 1. `s/jsx/md`?
2. Ditto my comment about linking to the template (or even embedding it if Github supports it?) so we won't have to maintain two sources of truth?

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Comment by derrandz** [comment]
This is a pending item before this gets merged:

- [ ] @derrandz to add guidelines (_suggestions_) on how to use Github Projects to coordinate contribution

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

File: docs/contributing/contribution/development.md
Comment (derrandz) [reply docs/contributing/contribution/development.md:88]: would be impracticable imo

File: docs/contributing/contribution/development.md
Comment (derrandz) [inline docs/contributing/contribution/development.md:174]: Let's be explicit by saying minimum time to re-request reviews is 2 days.

File: docs/contributing/contribution/development.md
Comment (derrandz) [inline docs/contributing/contribution/development.md:182]: will be required to pass all the of the Regression Tests Suite

File: docs/contributing/contribution/development.md
Comment (derrandz) [inline docs/contributing/contribution/development.md:183]: comments reviewed and **resolved**

**Review by derrandz — CHANGES_REQUESTED** [review]
Will continue with the review where I left off sometime later today.

File: docs/contributing/contribution/merge.md
Comment (guettomusick) [reply docs/contributing/contribution/merge.md:34]: Won't do, changing template MD for links

File: .github/ISSUE_TEMPLATE/issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/issue.md:6]: good question, let me check, but for sure not issue, maybe integration-issue

File: docs/contributing/CONTRIBUTING.md
Comment (guettomusick) [reply docs/contributing/CONTRIBUTING.md:15]: The links are on the sections below

File: .github/ISSUE_TEMPLATE/integration-issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/integration-issue.md:21]: Please check 9a53e83 and 7b5826e

**Comment by guettomusick** [comment]
@derrandz @Olshansk please take a look at 9a53e83 and 7b5826e, most of the comments are addressed on those commits

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (derrandz) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: Use `##` and `___` underneath it

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (derrandz) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Use `##` and `___` underneath it

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (derrandz) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Use `##` and `___` underneath it

**Review by Olshansk — APPROVED** [review]
Other than the comments left by @derrandz, lgtm!

File: .github/ISSUE_TEMPLATE/issue.md
Comment (guettomusick) [reply .github/ISSUE_TEMPLATE/issue.md:6]: I believe we can, but does it make sense? On the document, the assignee is the own working on the issue, so no assignee is an issue that hasn't started working yet.

**Comment by guettomusick** [comment]
@luyzdeleon @derrandz Your request change is blocking the merge, please review the final version to be able to merge.

@luyzdeleon I think your comment makes total sense for PR, but not so much for Issues, the thing is that PR template doesn't;t allow default assigned, I will check the settings to see if there's another way.

**Comment by guettomusick** [comment]
@luyzdeleon  I was able to add a CODEOWNER to enable default reviewers on all PRs

**Comment by guettomusick** [comment]
I also update the branch rule to enforce one review from a code owner to merge to a milestone and main, this will protect milestone branches and main branch

File: docs/contributing/contribution/development.md
Comment (guettomusick) [reply docs/contributing/contribution/development.md:150]: Any ideas on this? I think it will slow down the process. We can automate and send a notification on the goal chat when a PR is closed, will that work?

**Review by luyzdeleon — APPROVED** [review]
I'm satisfied with the changes made based on my comments, great job @guettomusick !

File: docs/contributing/contribution/development.md
Comment (derrandz) [reply docs/contributing/contribution/development.md:150]: I suppose it will, at least for the time being, will add to the backlong of ideas to discuss

File: docs/contributing/contribution/development.md
Comment (derrandz) [reply docs/contributing/contribution/development.md:110]: Just this one @guettomusick. Once this is changed please feel free to merge. I will approve for now.

**Review by derrandz — APPROVED** [review]
Make sure you address that one last change before merging. I will approve regardless. Thanks for the good work Fernando!

**Comment by guettomusick** [comment]
Reviewed and Approved by 3 contributors

**Pull request merged**
